calculator.rb

```diff
@@ -0,0 +1,52 @@
+def addition(first_number, second_number)
+  return first_number + second_number
+end
+def subtraction(first_number, second_number)
+  return first_number - second_number
+end
+def multiplication(first_number, second_number)
+  return first_number * second_number
+end
+def division(first_number, second_number)
+  return first_number / second_number
+end
+
+puts "Welcome to the calculator program. Which operator would you like to use?"
+puts "
+1. Add (+)
+2. Subtract (-)
+3. Multiply (*)
+4. Divide (/)
+Please choose your operator, you can input either the name or symbol:"
+operator = gets.chomp.downcase
+
+until operator == "add" || operator == "+" || operator == "subtract" || operator == "-" || operator == "multiply" || operator == "*" || operator == "divide" || operator == "/"
+  puts "Please enter a valid operator. Add (+), subtract (-), multiply (*), divide (/)."
+  operator = gets.chomp
+end
+
+puts "Please enter two numbers:"
+first_number = gets.chomp.to_i
+second_number = gets.chomp.to_i
+while first_number == 0 || second_number == 0
```

Comment:
I should be able to add or multiply a number by 0.  This doesn't completely work right.